src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -683,6 +687,10 @@ public void PreShutdown()
         {
             _logger.LogDebug($"Preventing any new worker processes from starting during shutdown.");
             _processStartCancellationToken.Cancel();
+            if (!_hostingConfigOptions.Value.RevertWorkerShutdownBehavior)
+            {
+                ShutdownWebhostLanguageWorkerChannels();
```

Comment:
Why is this the fix for the linked issue? Would like to see a description in the PR explaining what you've found, thanks.

As for this shutdown method - it's name is misleading since it only shuts down standby channels other than the channels for the configured worker runtime, and this should have already been done after initialization [here](https://github.com/Azure/azure-functions-host/blob/af3687ddf6fc1849cb954defb1e9e300552aac30/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L260). Should probably be renamed to ShutdownStandbyWorkerChannels to match the debounce delegate that is used.

Reply:
Looks like i had linked the wrong issue here. I have fixed the issue link, have added description on what I found through my investigation as well as the what the fix is trying to do. 

### For ShutdownStandbyWorkerChannels: 
`ShutdownWebhostLanguageWorkerChannels` is actually shutting down all the worker channels, not just the standby channels. The log statement here is actually misleading we may need to fix it. Also, I debugged through and for some reason the debounce delegate doesn't seem to trigger the underlying method.

https://github.com/Azure/azure-functions-host/blob/af3687ddf6fc1849cb954defb1e9e300552aac30/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L184-L188

https://github.com/Azure/azure-functions-host/blob/af3687ddf6fc1849cb954defb1e9e300552aac30/src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs#L334-L371